packages/snaps-controllers/src/snaps/endowments/name-lookup.ts

```diff
@@ -77,10 +116,46 @@ function validateCaveat(caveat: Caveat<string, any>): void {
 
   const { value } = caveat;
 
-  assert(
-    Array.isArray(value) && value.every((val) => isChainId(val)),
-    'Expected caveat value to have type "string array"',
-  );
+  switch (caveat.type) {
```

Comment:
Can we use Superstruct for this validation? We already have the structs for it 🤔